### DISCUSSION TITLE
Calculated column

### DISCUSSION THREAD

**Discussion by muezz** [ghiscoding/slickgrid-react#272] [Q&A] [answered]
Is it possible to have a column that is not editable (read only) but updates as referenced cell values are edited?

I suspect I could do it outside Slickgrid (manually) but I am wondering if this is supported or not.

**Comment by ghiscoding** [comment]
I'm not exactly sure what you mean, you should probably search on Stack Overflow for SlickGrid update and such.

**Reply by muezz** [reply]
I meant like a column that is calculated based on other columns and is not edit-able itself.

**Reply by ghiscoding** [reply, marked as answer]
I'm not sure if you're talking about column width calculation (see [docs](https://ghiscoding.gitbook.io/slickgrid-react/grid-functionalities/resize-by-cell-content)) or if you're talking about calculating 2 cell property and show its total on another cell on the right, if that's the case then just use a Custom Formatter, the 4th and 5th arguments are columnDef and dataContext (item), you have access to the entire row item (dataContext) so you can easily calculate stuff and then use Grouping if you need to calculate group totals.

```ts
const sumFormatter: Formatter = (row, cell, val, columnDef, dataContext) => {
  const total = dataContext['price'] * dataContext['qty']; // calculate
  dataContext[columnDef.field] = total; // assign
}
```

**Reply by muezz** [reply]
Yes. I did mean something like this where the data from server is just `col1` and `col2` but the UI (Slickgrid) shows 3 columns and the `col3` is just some aggregation (like sum) of the other columns.
